base/main.cpp

```diff
@@ -303,6 +303,9 @@ static Common::Error runGame(const Plugin *plugin, const Plugin *enginePlugin, O
 	// Inform backend that the engine is about to be run
 	system.engineInit();
 
+	// Purge all queued keyboard events that may remain from the GUI
+	system.getEventManager()->purgeKeyboardEvents();
```

Comment:
I wonder why only keyboard events are purged here? What about mouse and/or joystick? Wouldn't it make a lot more sense to wipe all those those input related events?

Reply:
I started with keyboard because those are the known problem and I can't vouch for the effects of anything else.

Purging others event types sounds reasonable but may someone familiar with the other Common::EventTypes should do that. There's often an EVENT_SCREEN_CHANGED in the queue, is removing that a problem? I have no idea.

Reply:
There's a `purgeMouseEvents()` which should be safe to add. Other inputs are probably safe too, but most engines probably don't have any special joystick/touch handling.

For clearing everything there's already `getEventDispatcher()->clearEvents()` but like @sluicebox pointed out, don't know which events we wanna keep.

Reply:
I'm comfortable updating this to also call purgeMouseEvents().

I could also add purgeJoystickEvents() to handle EVENT_JOYAXIS_MOTION, EVENT_JOYBUTTON_DOWN, EVENT_JOYBUTTON_UP if we want, but that's something I wouldn't be able to test. I think that plus purgeMouseEvents() would cover all the traditional input events.

What do you think? My goal is just to get the very real key-up bug safely fixed in time for release.

Reply:
Keyboard and mouse events are the most straightforward events to handle here, so they should be safe to be cleared in order to fix the issue at hand. Research for the other event types can be done post release